src/core/thread/mle.cpp

```diff
@@ -216,16 +247,19 @@ Error Mle::Start(StartMode aMode)
 #if OPENTHREAD_FTD
     else if (IsActiveRouter(GetRloc16()))
     {
-        if (Get<MleRouter>().BecomeRouter(ThreadStatusTlv::kTooFewRouters) != kErrorNone)
-        {
-            Attach(kAnyPartition);
-        }
+        mDeterminingLeader = false;
```

Comment:
This mDeterminingLeader thing was something I was trying out, but I had left set to false here for my testing, so it wasn't actually doing anything. Probably will want to remove it